packages/modeling/src/operations/booleans/trees/Node.js

```diff
@@ -99,28 +99,35 @@ class Node {
         current = stack.pop()
         continue
       }
+      let index = 0 // default
       if (!node.plane) {
-        let index = 0 // default
-        index = Math.floor(polygontreenodes.length / 2)
+        index = current.nextIndex || 0
```

Comment:
Please revert back as this will affect performance.

This will probably affect the tests again as well.

Reply:
this integral part of the fix. Previous code used middle point and hoped it would work ... and the same as with choosing any random point (I see in code sbdy also tried this) it will work only for some.

Th code must try other points or it will in some cases just be stuck in an infinite loop.

Reply:
If that’s the case then I really doubt the reliability of this fix. The starting point of the tree traversal shouldn’t matter.

These changes will have to be tested carefully.

Reply:
It does not matter, it would work the same if we start from the middle, but it matters because it complicates the stop condition (line 123 and line 130)

instead fo going from 0... length 
 - start from middle
 - go to end
 - wrap around to 0
 - go to middle-1 and then check if we need to stop splitting